lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -708,15 +708,15 @@ class SqlParser(private val ion: IonSystem) : Parser {
             }
             TIME -> {
                 val timeString = token!!.text!!
-                val precision = min(children[0].token?.value?.numberValue()?.toInt() ?: MAX_PRECISION_FOR_TIME, MAX_PRECISION_FOR_TIME)
+                val precision = children[0].token?.value?.numberValue()?.toInt()
```

Comment:
You probably want the `!!` here or better yet `?:` and throw in the null case something more meaningful.

Reply:
If precision is not specified, what is it?  I would assume you infer it from the literal, yes?

Reply:
1. Good catch. Thanks. It will never be `null` so I will add `!!`.
2. That's true. The precision is inferred from the literal in `SqlParser` during parsing.